tests/problems/test_tsp.py

```diff
@@ -0,0 +1,38 @@
+import math
+import pytest
+from evol.problems.tsp import TSPProblem
+
+
+def test_distance_func():
+    distances = TSPProblem.calc_distance_matrix([(0,0), (0,1), (1,0), (1,1)])
+    assert distances[0][0] == 0
+    assert distances[1][0] == 1
+    assert distances[0][1] == 1
+    assert distances[1][1] == 0
+    assert distances[0][2] == 1
+    assert distances[0][3] == math.sqrt(2)
```

Comment:
I guess it's better to use `pytest.approx` here, just in case.

Reply:
added that to a whole bunch of places. good point.